cmd/convertor/builder/builder.go

```diff
@@ -99,13 +117,49 @@ func (b *overlaybdBuilder) Build(ctx context.Context) error {
 		}
 	}()
 
+	var chain []digest.Digest
+	srcDiffIDs := b.config.RootFS.DiffIDs
+
 	for i := 0; i < b.layers; i++ {
 		downloaded[i] = make(chan error)
 		converted[i] = make(chan error)
+		alreadyConverted[i] = make(chan *v1.Descriptor)
+
+		chain = append(chain, srcDiffIDs[i])
+		chainID := identity.ChainID(chain).String()
+
+		// deduplication Goroutine
+		go func(idx int, chainID string) {
+			defer close(alreadyConverted[idx])
+			// try to find chainID -> converted digest conversion if available
+			desc, err := b.engine.CheckForConvertedLayer(ctx, idx, chainID)
+			if err != nil {
+				// in the event of failure fallback to regular process
+				return
+			}
+			alreadyConverted[idx] <- desc
+		}(i, chainID)
```

Comment:
In the earlier meeting this week I think the discussion noted that this wouldn't make much of a difference. I think it's a little bit faster to keep the existing approach and while it is a tad harder to follow since we are already using the pattern for download -> convert it seems alright to keep as is despite the added complexity.